### PR TITLE
Added INITIALISE subworkflow and cleaned up workflow main.nf

### PR DESCRIPTION
**Pull request by CarsonJM**
## PR checklist

- [X] This comment contains a description of changes (with reason).
- [X] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/phageannotator/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/phageannotator _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [X] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -42,24 +39,94 @@ WorkflowMain.initialise(workflow, params, log)
...
+include { PHAGEANNOTATOR } from './workflows/phageannotator/main'
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     NAMED WORKFLOW FOR PIPELINE
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 
-include { PHAGEANNOTATOR } from './workflows/phageannotator/main'
+include { paramsSummaryMap } from 'plugin/nf-validation'
```
Comment (sateeshperi) [inline main.nf]: do you need to include again when already done on L20 ?

File: workflows/phageannotator/main.nf [hunk 2]
```diff
@@ -71,39 +42,22 @@ def multiqc_report = []
 
 workflow PHAGEANNOTATOR {
 
+    take:
+    fastq_gz
+    fasta_gz
```
Comment (sateeshperi) [inline workflows/phageannotator/main.nf]:
```suggestion
    fasta_gz  // [ [ meta ], fasta.gz ]   , assemblies/genomes (mandatory)
```

Comment (sateeshperi) [reply workflows/phageannotator/main.nf]: recommend adding info about input about how the channel should like.....above is just an ex

File: workflows/phageannotator/main.nf [hunk 3]
```diff
@@ -71,38 +42,21 @@ def multiqc_report = []
 
 workflow PHAGEANNOTATOR {
 
+    take:
+    fastq_gz    // [ [ meta ], reads.fastq.gz ]     , reads (mandatory)
+    fasta_gz    // [ [ meta ], assembly.fastq.gz ]  , assemblies/genomes (mandatory)
```
Comment (edmundmiller) [inline workflows/phageannotator/main.nf]: I'm not sure how these are intended to be used, but if they're supposed to be a fasta linked to a sample it all needs to be in one channel.

File: subworkflows/local/fasta_virus_classification_genomad/tests/genomad_db_provided.nf.test.snap
Comment (edmundmiller) [inline subworkflows/local/fasta_virus_classification_genomad/tests/genomad_db_provided.nf.test.snap:1]: Should we add snaps to the `.gitattributes`

```
*.nf.test.snap linguist-generated
```

People can open them if they want to see the diff, but it's also extra files to mark as reviewed.

File: workflows/phageannotator/main.nf (hunk 3 above)
Comment (CarsonJM) [reply workflows/phageannotator/main.nf]: The fastq_gz channel contains reads, and the fasta_gz channel contains the assemblies (both from the same sample). Do you think I should make a sample that contains the reads and assemblies?

**Review by sateeshperi — APPROVED** [review]

**Pull request merged**